AdyenActions/AdyenActionComponent.swift

```diff
@@ -114,10 +102,10 @@ public final class AdyenActionComponent: ActionComponent, Localizable {
     }
     
     private func handle(_ action: ThreeDS2ChallengeAction) {
-        guard let threeDS2Component = threeDS2Component else { return }
+        guard let threeDS2Component = currentActionComponent as? ThreeDS2Component else { return }
```

Comment:
we should make an assertionFailure here in case this condition fails

Reply:
how does `Trying to handle a 3DS2Challenge, but current component is not a ThreeDS2Component` sound?

Reply:
sounds good to me